neqo-crypto/src/cert.rs

```diff
@@ -72,32 +74,26 @@ fn signed_cert_timestamp(fd: *mut PRFileDesc) -> Option<Vec<u8>> {
 
 impl CertificateInfo {
     pub(crate) fn new(fd: *mut PRFileDesc) -> Option<Self> {
-        peer_certificate_chain(fd).map(|(certs, cursor)| Self {
+        peer_certificate_chain(fd).map(|certs| Self {
             certs,
-            cursor,
             stapled_ocsp_responses: stapled_ocsp_responses(fd),
             signed_cert_timestamp: signed_cert_timestamp(fd),
         })
     }
+}
 
-    fn head(certs: &CertList) -> *const CERTCertListNode {
-        // Three stars: one for the reference, one for the wrapper, one to deference the pointer.
-        unsafe { addr_of!((***certs).list).cast() }
+impl CertificateInfo {
+    #[must_use]
+    pub fn iter(&self) -> ItemArrayIterator<'_> {
```

Comment:
I *think* that you can avoid a whole lot of work by using `null_safe_slice` here, using the `IntoIterator` implementation on the slice directly.

Reply:
I don't see it. I think we need the ItemArrayIterator wrapper to get the lifetimes right.